instructure/canvas-ios
# File module item publish form

### What's new?
- Added new file publish state icons to modules list to match web and the options of the permission dialog.
- Added form to control a file's permissions.

refs: MBL-17355
affects: Teacher
release note: none

test plan:
- Enable experimental feature flag in developer menu.
- Add a file to a module on web.
- Go to modules list and tap the kebab menu on the file's row.
- Check the editor dialog if it correctly prefills with the file's permission values.
- Modify file permissions and check if the displayed icon on the modules list matches the one on web.

## Screenshots

<table>
<tr>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/cae6ba35-ca35-482e-9242-0a6b325d8e44" maxHeight=500></td>
<td><img src="https://github.com/instructure/canvas-ios/assets/72396990/c97e7728-d015-46c6-88fd-f14053e1ae55" maxHeight=500></td>
</tr>
</table>

## Checklist
